blocksync/reactor.go

```diff
@@ -240,7 +240,15 @@ func (bcR *Reactor) Receive(e p2p.Envelope) {
 // NOTE: Don't sleep in the FOR_LOOP or otherwise slow it down!
 func (bcR *Reactor) poolRoutine(stateSynced bool) {
 	bcR.metrics.Syncing.Set(1)
-	defer bcR.metrics.Syncing.Set(0)
+	defer func() {
```

Comment:
If we set the metric back to 0 before deregistering will we capture the end of block sync in prometheus? If we want to query for running periods, this is easier than figuring out when metrics stopped being updated.

Reply:
I tried setting the `Syncing` metric to 0 before deregistering but it doesn't work. Prometheus doesn't have time to see the change to 0 before unregistering the metric. 

The advantage of not changing to 0 is that the graph for this metric is cleaner because it doesn't have to display the 0 values. And for querying running periods, there is the option of comparing whether it's equal to 1 or different from `nil`, instead of checking that it's not equal to 0. But if you think it's better to set it to 0, we could do just that; it's not a big difference.

Reply:
For the other metrics, I think it's necessary to unregister them. For example, this is what we would see if we want to display `cometbft_consensus_height` and `cometbft_blocksync_latest_block_height` at the same time. The red line for the height during syncing would get a permanent fixed value even after syncing has finished.
<img width="1132" alt="Screenshot 2023-03-17 at 15 27 45" src="https://user-images.githubusercontent.com/15466498/226316534-41afd386-a956-4a5b-bfd8-751b90f5a655.png">

Reply:
> I tried setting the `Syncing` metric to 0 before deregistering but it doesn't work. Prometheus doesn't have time to see the change to 0 before unregistering the metric.
> 
> The advantage of not changing to 0 is that the graph for this metric is cleaner because it doesn't have to display the 0 values. And for querying running periods, there is the option of comparing whether it's equal to 1 or different from `nil`, instead of checking that it's not equal to 0. But if you think it's better to set it to 0, we could do just that; it's not a big difference.

I see. As long as we can query, it LGTM.